CONTRIBUTING.md

```diff
@@ -0,0 +1,149 @@
+
+1. Focus debugging efforts on a simple code snippet,
+2. Ensure that the problem is not caused by unrelated dependencies/configuration,
+3. Avoid the need to share production codebases.
+
+#### Are Minimal Reproductions Required?
+
+In certain cases, creating a minimal reproduction might not be practical (e.g. due to nondeterministic factors, external dependencies). In such cases you would be asked to provide as much information as possible, for example by sharing a memory dump of the failing application. If maintainers are unable to root cause the problem, they might still close the issue as not actionable. While not required, minimal reproductions are strongly encouraged and will significantly improve the chances of your issue being prioritized and fixed by the maintainers.
+
+#### How to Create a Minimal Reproduction
+
+The best way to create a minimal reproduction is gradually removing code and dependencies from a reproducing app, until the problem no longer occurs. A good minimal reproduction:
+
+- Excludes all unnecessary types, methods, code blocks, source files, nuget dependencies and project configurations.
+- Contains documentation or code comments illustrating expected vs actual behavior.
+- If possible, avoids performing any unneeded IO or system calls.
+
+## Contributing Changes
+
+Project maintainers will merge changes that improve the product significantly.
+
+Maintainers will not merge changes that have narrowly-defined benefits, due to compatibility risk. Companies are building products on top of Enmeshed. We may revert changes if they are found to be breaking.
+
+### DOs and DON'Ts
+
+Please do:
+
+- **DO** follow our coding style. These are enforced by prettier as well as ESlint, which are executed by our CI-pipelines.
+- **DO** include tests when adding new features. When fixing bugs, start with adding a test that highlights how the current behavior is broken.
+- **DO** keep the discussions focused. When a new or related topic comes up it's often better to create new issue than to side track the discussion.
+- **DO** blog and tweet (or whatever) about your contributions, frequently!
+
+Please do not:
+
+- **DON'T** make PRs for style changes.
+- **DON'T** surprise us with big pull requests. Instead, file an issue and start a discussion so we can agree on a direction before you invest a large amount of time.
+- **DON'T** commit code that you didn't write. If you find code that you think is a good fit to add to Enmeshed, file an issue and start a discussion before proceeding.
+- **DON'T** submit PRs that alter licensing related files or headers. If you believe there's a problem with them, file an issue and we'll be happy to discuss it.
+
+### Breaking Changes
+
+Contributions must maintain API signature and behavioral compatibility. Contributions that include breaking changes will be rejected. Please file an issue to discuss your idea or change if you believe that it may affect compatibility.
```

Comment:
"Contributions that include breaking changes will be rejected." Ich würde das nicht so harsch sehen, sondern eher "Contributions that include breaking changes will be rejected for the current releases and can only be considered for the next major release."